planet/src/main/java/com/google/planet/data/ItineraryOrder.java

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.planet.data;
+
+public final class ItineraryOrder {
```

Comment:
if you wanted, this could be an inner class of ItineraryGenerator. But this is works too. It's a stylistic choice

Reply:
Okay!

Reply:
Not sure why I can't reply to the above comment so i'm replying to it here.

When the user inputs an invalid address, the getDirectionRoute will fail and the errorMessage will be set
I wanted to try using an exception, however I'm not sure how to bubble up the errors since many functions are nested

Reply:
I'm not sure I follow why nested functions would prevent us from using exceptions. They'll bubble up the stack until they are caught (or until the stack unwinds and the app crashes). 

Generally, the recommendation would be to construct an exception with the appropriate error message and then catch it right before you construct the final response to the user. That way, you can construct an appropriate response to tell the user what to fix

Reply:
You're right... I just changed all the error messages to exceptions